### PR TITLE
Add tests to issue/burn more than INT64_MAX in a single path payment

### PR DESCRIPTION
**Pull request by sisuresh**
# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/transactions/test/PathPaymentTests.cpp [hunk 1]
```diff
@@ -112,6 +112,78 @@ TEST_CASE("pathpayment", "[tx][pathpayment]")
...
+        a1.changeTrust(cur2, trustLineLimit);
+
+        a2.changeTrust(cur2, trustLineLimit);
+        a2.changeTrust(cur1, trustLineLimit);
+
+        a3.changeTrust(idr, trustLineLimit);
+        a3.changeTrust(cur2, trustLineLimit);
+        a3.changeTrust(cur1, trustLineLimit);
+
+        SECTION("issue more than INT64_MAX")
```
Comment (jonjove) [inline src/transactions/test/PathPaymentTests.cpp:131]: I'm not seeing how this shows that more than `INT64_MAX` was issued. Unless I'm missing something, it looks to me like you have demonstrated issuing `INT64_MAX` and transferring `INT64_MAX/2`. This is interesting in its own right, but it's not what the test claims.

File: src/transactions/test/PathPaymentTests.cpp [hunk 2]
```diff
@@ -112,6 +112,78 @@ TEST_CASE("pathpayment", "[tx][pathpayment]")
...
+
+                // CUR2 -> IDR -> CUR2 -> CUR1
+                gateway.pay(a3, cur2, INT64_MAX, cur1, INT32_MAX, {idr, cur2});
+
+                REQUIRE(INT64_MAX - a1.getTrustlineBalance(cur2) <
+                        a2.getTrustlineBalance(cur2));
+            });
+        }
+
+        SECTION("burn more than INT64_MAX")
```
Comment (jonjove) [inline src/transactions/test/PathPaymentTests.cpp:158]: Maybe I'm missing something, but this also doesn't seem to be burning more than `INT64_MAX`? I see a similar situation to the other section.

**Review by jonjove — COMMENTED** [review]
I have two thoughts here:
1. I'm not sure you're testing what you claim to be testing.
2. Comments would be really helpful, particularly annotating what the purpose of each order is.

File: src/transactions/test/PathPaymentTests.cpp (hunk 2 above)
Comment (sisuresh) [reply src/transactions/test/PathPaymentTests.cpp]: Good catch. I was looking at the test as the issuer sending/receiving over INT64_MAX, but I didn't take into account the intermediate hop that did the opposite. After thinking about this, I realized I was making the test much more complicated than it needed to be. The updated test uses a path payment from `a1` to hit offers set up by the issuer.

**Comment by jonjove** [comment]
r+ 2e3b7cb

**Pull request merged**